File: src/utils/paths.js
"use strict";

const os = require("os");
const path = require("path");

// let pwd = process.env.PWD; // [https://stackoverflow.com/a/39740187]
// [https://stackoverflow.com/a/29496638]
let cwd = path.dirname(path.dirname(__dirname));
let homedir = os.homedir();
let projectname = "nodecliac";
let ncliacdir = path.join(homedir, `.${projectname}`);
let bashrcpath = path.join(homedir, ".bashrc");
let mainscriptname = "init.sh";
let registrypath = path.join(homedir, `.${projectname}`, "registry");
let acmapssource = path.join(homedir, `.${projectname}`, "src");
let setupfilepath = path.join(ncliacdir, `.setup.db.json`);
let resourcespath = path.join(cwd, "resources", "packages");
let resourcessrcs = path.join(cwd, "src", "scripts");
let testsrcpath = path.join(cwd, "tests", "scripts");
let cachepath = path.join(homedir, `.${projectname}`, ".cache");
let cachelevel = path.join(homedir, `.${projectname}`, ".cache-level");

module.exports = {
	paths: {
		cwd,
		homedir,
		ncliacdir,
		bashrcpath,
		mainscriptname,
		registrypath,
		acmapssource,
		setupfilepath,
		resourcespath,
		resourcessrcs,
		testsrcpath,
		cachepath,
		cachelevel
	}
};

File: src/parser/nim/utils/paths.nim
import os, strformat, tables

let cwd = parentDir(parentDir(currentSourcePath()))
let homedir = os.getEnv("HOME")
const projectname = "nodecliac"
let ncliacdir = joinPath(homedir, fmt".{projectname}")
let bashrcpath = joinPath(homedir, ".bashrc")
const mainscriptname = "init.sh"
let registrypath = joinPath(homedir, fmt".{projectname}", "registry")
let acmapssource = joinPath(homedir, fmt".{projectname}", "src")
let setupfilepath = joinPath(ncliacdir, ".setup.db.json")
let resourcespath = joinPath(cwd, "resources", "packages")
let resourcessrcs = joinPath(cwd, "src", "scripts")
let testsrcpath = joinPath(cwd, "tests", "scripts")
let cachepath = joinPath(homedir, fmt".{projectname}", ".cache")

let paths* = {
    "cwd": cwd,
    "homedir": homedir,
    "ncliacdir": ncliacdir,
    "bashrcpath": bashrcpath,
    "mainscriptname": mainscriptname,
    "registrypath": registrypath,
    "acmapssource": acmapssource,
    "setupfilepath": setupfilepath,
    "resourcespath": resourcespath,
    "resourcessrcs": resourcessrcs,
    "testsrcpath": testsrcpath,
    "cachepath": cachepath
}.toTable
